### PR TITLE
Provision A records

### PR DESCRIPTION
**Pull request by jvatic**
ping @titanous

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -210,6 +223,84 @@ func (a *API) ProvisionDomain(w http.ResponseWriter, req *http.Request, params h
...
+	if len(reqData.IPAddresses) < 1 {
+		hh.Error(w, hh.JSONError{
+			Code:    hh.ValidationError,
+			Message: "Must provide name servers or IPv4 addresses",
+		})
+		return
+	}
+	for _, a := range reqData.Nameservers {
+		for _, p := range strings.Split(a, ".") {
+			i := strconv.Atoi(p)
```
Comment (titanous) [inline server.go]: Do `net.ParseIP(a).To4()` and check that it isn't nil.

File: server.go [hunk 2]
```diff
@@ -210,6 +223,84 @@ func (a *API) ProvisionDomain(w http.ResponseWriter, req *http.Request, params h
...
+						Name:            aws.String(domain + "."),
+						TTL:             aws.Long(3600),
+						Type:            aws.String(route53.RRTypeA),
+						ResourceRecords: records,
+					},
+				},
+				{
+					Action: aws.String(route53.ChangeActionCreate),
+					ResourceRecordSet: &route53.ResourceRecordSet{
+						Name:            aws.String("*." + domain + "."),
```
Comment (titanous) [inline server.go]: Use `fmt.Sprintf`

File: server.go [hunk 3]
```diff
@@ -165,7 +178,7 @@ func (a *API) ProvisionDomain(w http.ResponseWriter, req *http.Request, params h
 
 	nsData, _ := json.Marshal(reqData.Nameservers)
 	const nsUpdate = "UPDATE domains SET nameservers = $2 WHERE domain = $1 AND nameservers IS NULL"
-	updateRes, err := a.db.Exec(nsUpdate, domain, nsData)
+	updateRes, err := a.db.Exec(nsUpdate, reqData.Domain, nsData)
```
Comment (titanous) [inline server.go:180]: This also needs to check that `ip_addresses` is NULL.

File: server.go [hunk 4]
```diff
@@ -210,6 +223,84 @@ func (a *API) ProvisionDomain(w http.ResponseWriter, req *http.Request, params h
...
+					Code:    hh.ValidationError,
+					Message: "Invalid IPv4 addresses provided",
+				})
+				return
+			}
+		}
+	}
+
+	ipData, _ := json.Marshal(reqData.IPAddresses)
+	const ipUpdate = "UPDATE domains SET ip_addresses = $2 WHERE domain = $1 AND ip_addresses IS NULL"
```
Comment (titanous) [inline server.go]: This also needs to check that `nameservers` is NULL.

**Comment by jvatic** [comment]
@titanous comments addressed

**Comment by titanous** [comment]
LGTM

**Pull request merged**